src/materialized/src/bin/materialized/sys.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! System support functions.
+
+use log::{warn, trace};
+
+#[cfg(not(any(target_os = "macos", target_os = "linux", target_os = "ios")))]
+pub fn adjust_rlimits() {
+    trace!("rlimit crate does not support this OS; not adjusting nofile limit");
+}
+
+/// Attempts to increase the soft nofile rlimit to the maximum possible value.
+#[cfg(any(target_os = "macos", target_os = "linux", target_os = "ios"))]
+pub fn adjust_rlimits() {
+    use rlimit::{Resource, Rlim};
+
+    // getrlimit/setrlimit can have surprisingly different behavior across
+    // platforms, even with the rlimit wrapper crate that we use. This function
+    // is chattier than normal at the trace log level in an attempt to ease
+    // debugging of such differences.
+
+    let (soft, hard) = match Resource::NOFILE.get() {
+        Ok(limits) => limits,
+        Err(e) => {
+            trace!("unable to read initial nofile rlimit: {}", e);
+            return;
+        }
+    };
+    trace!("initial nofile rlimit: ({}, {})", soft, hard);
+
+    #[cfg(target_os = "macos")]
+    let hard = {
+        use ore::cast::CastFrom;
+        use sysctl::Sysctl;
+
+        // On macOS, getrlimit by default reports that the hard limit is
```

Comment:
Is there a source for this comment? I don't doubt it at all but this is the sort of thing that if I was reading a year from now I'd wonder how the author knew that

Reply:
https://github.com/cockroachdb/cockroach/commit/e777ba081ac1fd9930420aecdb1638e816894cb0 :D